sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -325,8 +328,10 @@ object ScalaReflection extends ScalaReflection {
         }
 
         val companion = t.normalize.typeSymbol.companionSymbol.typeSignature
-        val cls = companion.declaration(newTermName("newBuilder")) match {
-          case NoSymbol => classOf[Seq[_]]
+        val cls = companion.member(newTermName("newBuilder")) match {
```

Comment:
It works.

    To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
    Welcome to
          ____              __
         / __/__  ___ _____/ /__
        _\ \/ _ \/ _ `/ __/  '_/
       /___/ .__/\_,_/_/ /_/\_\   version 2.3.0-SNAPSHOT
          /_/

    Using Scala version 2.10.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_102)
    Type in expressions to have them evaluated.
    Type :help for more information.
    ...

    scala> val t = ScalaReflection.localTypeOf[scala.collection.mutable.HashSet[Int]]
    t: org.apache.spark.sql.catalyst.ScalaReflection.universe.Type = scala.collection.mutable.HashSet[Int]

    scala> val companion = t.normalize.typeSymbol.companionSymbol.typeSignature
    companion: org.apache.spark.sql.catalyst.ScalaReflection.universe.Type = scala.collection.mutable.HashSet.type

    scala> val cls = companion.member(newTermName("newBuilder"))
    cls: org.apache.spark.sql.catalyst.ScalaReflection.universe.Symbol = method newBuilder